subprojects/core/src/main/java/org/gradle/process/internal/util/LongCommandLineDetectionUtil.java

```diff
@@ -25,6 +25,8 @@
     public static final int MAX_COMMAND_LINE_LENGTH_WINDOWS = 32767;
     // Derived from default when running getconf ARG_MAX in OSX
     public static final int MAX_COMMAND_LINE_LENGTH_OSX = 262144;
+    // Dervied from MAX_ARG_STRLEN as per http://man7.org/linux/man-pages/man2/execve.2.html
+    public static final int MAX_COMMAND_LINE_LENGTH_LINUX = 131072;
     public static final int MAX_COMMAND_LINE_LENGTH_NIX = 2097152;
```

Comment:
I suspect this value should be changed rather than using another one. Are there any other nix variants where 2097152 is appropriate?

Reply:
I don't know. I don't have familiarity with Solaris and other platforms. But since OSX is different than Linux, I didn't feel comfortable changing the default.

Reply:
[Actual values for ARG_MAX (or NCARGS)](https://www.in-ulm.de/~mascheck/various/argmax/#results) is an older document that looks comprehensive. tl;dr no. it varies a lot.

Reply:
Basically all of those are lower than 2097152. Since it shouldn't typically hurt to have /too low/ of a value, defaulting to 131072 seems like it would at least improve the situation.